Token.js

```diff
@@ -2166,15 +2158,13 @@ class Token {
 					};
 				
 				
-					let canvas = document.getElementById("raycastingCanvas");
-					let ctx = canvas.getContext("2d", { willReadFrequently: true });
 
 					if(!window.DM && window.playerTokenAuraIsLight){
 						const left = (tokenPosition.x + (parseFloat(self.options.size) / 2)) / parseFloat(window.CURRENT_SCENE_DATA.scale_factor);
 						const top = (tokenPosition.y + (parseFloat(self.options.size) / 2)) / parseFloat(window.CURRENT_SCENE_DATA.scale_factor);
 						if(typeof left != 'number' || isNaN(left) || typeof top != 'number' || isNaN(top)){
 							showErrorMessage(
-							  Error(`One of these values is not a number: Size: ${self.options.size}, Scene Scale: ${window.CURRENT_SCENE_DATA.scale_factor}, x: ${tokenPosition.x}, y: ${tokenPosition.y}`),
+							  Error(`One of these values is not a number: Size: ${self.options.size}, Scene Scale: ${window.CURRENT_SCENE_DATA.scale_factor}, x: ${tokenPosition.x}, y: ${tokenPosition.y}, pagex: ${event.pageX}, clickx: ${click.x}, originalleft: ${original.left}, pageY: ${event.pageY}, clickY: ${click.y}, original.top: ${original.top}, zoom: ${zoom}`),
```

Comment:
I wonder if in our goal to reduce memory usage we are accidentally releasing one of these out of scope making them undefined

Reply:
I'll add the above ones, my thought process on the window variables was if it was those it would happen to all tokens not just some but you're right in that it's better to be sure - we already have some window variables there anyway. Also adding `window.CURRENT_SCENE_DATA.offsety` and `window.CURRENT_SCENE_DATA.offsetx` in case its that in the snap to grid function. It's a lot but really we're the only ones who will probably decipher the error anyway lol.

I think the only thing that could be released during the start/drag is the click.y and click.x as everything else comes from the token data or window variables. It's certainly possible though.